### ISSUE TITLE
[Improvement]: Create an interface for the `ClassDefinition`

### ISSUE THREAD

**Issue by Corepex** [pimcore/pimcore#15429]
### Improvement description

Since the `ClassDefinition` is a final class we can't mock the class for unit tests.

https://github.com/pimcore/pimcore/blob/11.x/models/DataObject/ClassDefinition.php#L38

**Comment by jdreesen** [comment]
This is what #14357 already does, right?

**Comment by Corepex** [comment]
> This is what #14357 already does, right?

That's correct, but we need it to finish some other tasks, and therefore we "cherry-picked" the needed changes.
